mono/metadata/icall.c

```diff
@@ -2163,25 +2163,39 @@ ves_icall_MonoField_SetValueInternal (MonoReflectionFieldHandle field, MonoObjec
 		mono_gchandle_free (value_gchandle);
 }
 
+ICALL_EXPORT MonoObject*
+mono_TypedReference_ToObject (MonoTypedRef* tref);
+
 ICALL_EXPORT void
 ves_icall_System_RuntimeFieldHandle_SetValueDirect (MonoReflectionField *field, MonoReflectionType *field_type, MonoTypedRef *obj, MonoObject *value, MonoReflectionType *context_type)
 {
+	HANDLE_FUNCTION_ENTER ();
+
+	ERROR_DECL (error);
+
 	MonoClassField *f;
+	MonoObjectHandle objHandle, valueHandle;
+	MonoReflectionFieldHandle fieldHandle;
 
 	g_assert (field);
 	g_assert (obj);
 	g_assert (value);
 
 	f = field->field;
 	if (!MONO_TYPE_ISSTRUCT (m_class_get_byval_arg (f->parent))) {
-		mono_set_pending_exception (mono_get_exception_not_implemented (NULL));
-		return;
+		fieldHandle = MONO_HANDLE_NEW (MonoReflectionField, field);
+		objHandle = MONO_HANDLE_NEW (MonoObject, mono_TypedReference_ToObject (obj));
+		valueHandle = MONO_HANDLE_NEW (MonoObject, value);
```

Comment:
For that matter, you can pass temporaries, don't need the local variables.

Reply:
Makes sense, I thought they needed to be in scope on the stack but knowing what I do now about setup/teardown for coop I can understand why this is safe.

Reply:
There is also a shorthand available using token pasting, like if you name the pointers foo_raw then you can get handles foo. HANDLE_DCL or such. But I think best is just to pass temporaries and have no locals.